driver/utils/unicode_conv.h

```diff
@@ -11,6 +11,20 @@
 
 #include <cstring>
 
+class UnicodeConversionContext {
+public:
+    UnicodeConversionContext() {} // ...to call explicit c-tors of member objects.
+
+public:
+//  std::locale source_locale;
+//  std::locale destination_locale;
+#if !defined(_MSC_VER) || _MSC_VER >= 1920
+    std::wstring_convert<std::codecvt_utf8<char16_t>, char16_t> UCS2_converter_char16;
+    std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t> UCS2_converter_char32;
+#endif
+    std::wstring_convert<std::codecvt_utf8<wchar_t>, wchar_t> UCS2_converter_wchar;
+};
+
 using CharTypeLPCTSTR = std::remove_cv<std::remove_pointer<LPCTSTR>::type>::type;
 
 inline std::size_t NTSStringLength(const char * src, const std::locale& locale) {
```

Comment:
Why lots of `NTSStringLength` overloads do not utilize context, but still instantiate `std::codecvt_utf8` on the stack ?

Reply:
They are used in one place, which is not on a critical path.